src/main/java/org/threadly/concurrent/SingleThreadScheduler.java

```diff
@@ -52,33 +72,25 @@ public SingleThreadScheduler(boolean daemonThread) {
    * @param threadFactory factory to make thread for scheduler
    */
   public SingleThreadScheduler(ThreadFactory threadFactory) {
-    ArgumentVerifier.assertNotNull(threadFactory, "threadFactory");
-    
-    sManager = new AtomicReference<SchedulerManager>(null);
-    this.threadFactory = threadFactory;
+    this(null, DEFAULT_LOW_PRIORITY_MAX_WAIT_IN_MS, threadFactory);
   }
   
   /**
-   * Returns the {@link SchedulerManager} instance that contains the scheduler.  This does no 
-   * safety checks on if the scheduler is running or not, it just ensures that the single instance 
-   * of the manager is returned.
+   * Constructs a new {@link SingleThreadScheduler}.  No threads will start until the first task 
+   * is provided.
    * 
-   * @return Single instance of the SchedulerManager container
+   * 
+   * @param defaultPriority Default priority for tasks which are submitted without any specified priority
+   * @param maxWaitForLowPriorityInMs time low priority tasks to wait if there are high priority tasks ready to run
+   * @param threadFactory factory to make thread for scheduler
    */
-  private SchedulerManager getSchedulerManager() {
```

Comment:
In the SingleThreadScheduler this is no longer lazily constructed.  It is constructed at the begining, but is only started once we execute a task on to it still.